editor/modes/visual-editor/index.js

```diff
@@ -11,16 +17,48 @@ import Inserter from '../../inserter';
 import VisualEditorBlockList from './block-list';
 import PostTitle from '../../post-title';
 
-export default function VisualEditor() {
-	return (
-		<div
-			role="region"
-			aria-label={ __( 'Visual Editor' ) }
-			className="editor-visual-editor"
-		>
-			<PostTitle />
-			<VisualEditorBlockList />
-			<Inserter position="top right" />
-		</div>
-	);
+class VisualEditor extends Component {
+	constructor() {
+		super( ...arguments );
+		this.bindContainer = this.bindContainer.bind( this );
+		this.onClick = this.onClick.bind( this );
+	}
+
+	bindContainer( ref ) {
+		this.container = ref;
+	}
+
+	onClick( event ) {
+		if ( event.target === this.container ) {
```

Comment:
Between this and #836, I think I like this approach better. But it also means we probably don't need [the `stopPropagation` anymore](https://github.com/WordPress/gutenberg/blob/c1afcdb5bfcf5825f4908a55fccc0c7e3974f7c4/editor/modes/visual-editor/block.js#L157-L159).